arangod/Replication/Syncer.cpp

```diff
@@ -329,10 +329,20 @@ void Syncer::JobSynchronizer::request(std::function<void()> const& cb) {
     
 /// @brief notifies that a job was posted
 void Syncer::JobSynchronizer::jobPosted() {
-  CONDITION_LOCKER(guard, _condition);
-
-  TRI_ASSERT(_jobsInFlight == 0);
-  ++_jobsInFlight;
+  while (true) {
```

Comment:
check is syncer is beeing stopped ?!
```suggestion
  while (!_syncer->isAborted()) {
```